pkgs/by-name/op/opencode/package.nix

```diff
@@ -1,60 +1,137 @@
 {
   lib,
-  fetchFromGitHub,
+  stdenv,
   buildGoModule,
-  versionCheckHook,
+  bun,
+  fetchFromGitHub,
+  fetchurl,
   nix-update-script,
+  testers,
 }:
 
-buildGoModule (finalAttrs: {
-  pname = "opencode";
-  version = "0.0.52";
-
+let
+  version = "0.1.162";
   src = fetchFromGitHub {
     owner = "sst";
     repo = "opencode";
-    tag = "v${finalAttrs.version}";
-    hash = "sha256-wniGu8EXOI2/sCI7gv2luQgODRdes7tt1CoJ6Gs09ig=";
+    tag = "v${version}";
+    hash = "sha256-GNF7SWMh9FZjmL3EV0pT6gsR3HeEWZ+SpNWaParRwho=";
```

Comment:
Why did you move away from `finalAttrs`? It makes overriding easier.

Reply:
Ok my bad. Nevermind.

Reply:
Both `version` and `src` are defined in a `let` expression since they are shared across all three declarations (`opencode-tui`, `opencode-node-modules`, and `opencode`). I don't know how I could solve this differently.